### ISSUE TITLE
backport: Refactor the aggregation and The IN operator to multithread

### ISSUE THREAD

**Issue by hustjieke** [stoneatom/stonedb#1419]
**Is your feature request related to a problem? Please describe.**

previous working issue: #472, #422, #465, #466
commit id: 208f79d80bae1667d147e0713950317c9ca6c6b9
commit id: 0a176af5ce9820e56fa787a8f08f6427b459b224

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by hustjieke** [comment]
After backport, some resulst is wrong and increase some crashes, check it in next milestone.

**Comment by RingsC** [comment]
> After backport, some resulst is wrong and increase some crashes, check it in next milestone.

Yes, there are some potential issues in this part, and it should be reviewed carefully.
